module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -530,19 +618,31 @@ protected function getNewSitemapIndex()
      */
     protected function getNewSitemap()
     {
-        return new Sitemap($this->frequency);
+        $sitemap = new Sitemap($this->frequency);
+        if ($this->config->Sitemap->indexLanguageVersions) {
```

Comment:
This logic is potentially going to get repeated many times; should we do the work once and store it in a property?

Reply:
Done.